src/sonic-py-common/tests/device_info_test.py

```diff
@@ -51,9 +53,16 @@
 
 
 class TestDeviceInfo(object):
-    @classmethod
-    def setup_class(cls):
-        print("SETUP")
+    @pytest.fixture(scope="class", autouse=True)
+    def sanitize_environment(self):
+        # Back up environment variables then clear them, in case an environment
+        # variable is set in the test environment (e.g., PLATFORM) which could
+        # modify the behavior of sonic-py-common
+        env_orig = os.environ.copy()
+        os.environ.clear()
+        yield
+        # Restore original environment variables
+        os.environ.update(env_orig)
```

Comment:
are you only modify your own environ, why do you need to restore it? but again, i feel we are creating some virtual env here.

Reply:
Correct, there is really no need to restore the old environment, it's just a best practice to leave things as you found them.